autopilot/autopilot.go

```diff
@@ -389,20 +382,14 @@ func (ap *Autopilot) blockUntilConfigured(interrupt <-chan time.Time) (configure
 	var once sync.Once
 
 	for {
-		// try and fetch the config
-		ctx, cancel := context.WithTimeout(ap.shutdownCtx, 30*time.Second)
-		_, err := ap.bus.Autopilot(ctx, ap.id)
-		cancel()
-
-		// if the config was not found, or we were unable to fetch it, keep blocking
-		if utils.IsErr(err, context.Canceled) {
-			return
-		} else if utils.IsErr(err, api.ErrAutopilotNotFound) {
-			once.Do(func() { ap.logger.Info("autopilot is waiting to be configured...") })
-		} else if err != nil {
+		state, err := ap.bus.AutopilotState(ap.shutdownCtx)
+		if err != nil && !utils.IsErr(err, api.ErrAutopilotStateNotFound) {
 			ap.logger.Errorf("autopilot is unable to fetch its configuration from the bus, err: %v", err)
 		}
-		if err != nil {
+
+		configured = state.AutopilotConfig.Hosts != (api.HostsConfig{}) && state.AutopilotConfig.Contracts != (api.ContractsConfig{})
```

Comment:
I'd honestly be explicit about it and have an `Enabled` column. The advantage is that you get to keep your settings when you disable the autopilot in case you just disable it temporarily and we can initialize the autopilot config with sane defaults.

I'd also make it so that you can only change individual settings.

Reply:
Ok, I wasn't too sure how to interpret the last part of your comment but just took a stab at it. The number of routes in the `bus` are on the excessive side now but otoh it felt quite natural when I was updating all of the integration tests and so on.

```go
"GET    /autopilot"
"PUT    /autopilot/config/contracts"
"PUT    /autopilot/config/hosts"
"PUT    /autopilot/enable"
"PUT    /autopilot/period"
```

Reply:
I'd probably even reduce it to `PUT /autopilot` and have a json request like
```json
{
  "contracts": {...},
  "hosts": {...},
  "enabled": true|false,
}
and then figure out which of the fields were set. Similar to how we already do it for settings.
The `period` is probably fine to keep separate since it's not really config related.

@n8maninger @alexfreska do you have any preferences?

Reply:
Could combine it but then I'd propose:

```
GET /autopilot
PUT /autopilot/config
PUT /autopilot/period
```

Often clients use Apollo + GraphQL to combine http requests like this, not sure what i prefer really. In this case I'm leaning towards the reduced number of routes though.

Reply:
I agree with Alex